tests/spring-cloud-sleuth-instrumentation-reactor-tests/src/test/java/org/springframework/cloud/sleuth/instrument/web/client/WebClientBraveTests.java

```diff
@@ -107,14 +99,12 @@ public void onSubscribe(Subscription s) {
 			}
 
 			@Override
-			public void onNext(ClientResponse t) {
-				Subscription s = ref.getAndSet(null);
-				if (s != null) {
+			public void onNext(ClientResponse clientResponse) {
+				if (ref.getAndSet(null) != null) {
 					callback.onSuccess(null);
-					s.cancel();
```

Comment:
the other test doesn't call cancel, but I noticed sometimes this is called. This causes a problem really, if we assume that cancel is an error. For example, a success response might not happen due to this cancel depending on order cc @simonbasle @robotmrv

Reply:
`s` is the object that produced the `ClientResponse`, the `Subscription` generated by the `request` Mono, and if we're in `onNext` it means that it already produced it... so `s.cancel()` doesn't bring much _for that particular case_.

if the test could run on an arbitrary source, then `s.cancel()` would make sense to signal we're only interested in the first incoming element.

Reply:
thanks.. moved the thing about arbitrary source here https://github.com/spring-cloud/spring-cloud-sleuth/issues/1570#issuecomment-591247178